dmscripts/notify_suppliers_of_new_fw_cq_answers.py

```diff
@@ -0,0 +1,81 @@
+from logging import Logger
+from typing import Optional
+from uuid import UUID, uuid4
+
+from dmapiclient import DataAPIClient
+
+from dmutils.email import DMNotifyClient
+from dmutils.email.exceptions import EmailError
+from dmutils.email.helpers import hash_string
+from dmutils.env_helpers import get_web_url_from_stage
+
+
+def notify_suppliers_of_new_fw_cq_answers(
+    data_api_client: DataAPIClient,
+    notify_client: DMNotifyClient,
+    notify_template_id: str,
+    framework_slug: str,
+    stage: str,
+    dry_run: bool,
+    logger: Logger,
+    run_id: Optional[UUID] = None,
+) -> int:
+    run_is_new = bool(run_id)
+    run_id = run_id or uuid4()
+    logger.info(f"{'Starting' if run_is_new else 'Resuming'} run id {{run_id}}", extra={"run_id": str(run_id)})
+
+    framework = data_api_client.get_framework(framework_slug)["frameworks"]
+
+    if framework["status"] != "open":
+        raise ValueError(f"Framework {framework_slug!r} is not open (status is {framework['status']!r})")
+
+    failure_count = 0
+
+    for supplier_framework in data_api_client.find_framework_suppliers_iter(framework_slug):
+        for user in data_api_client.find_users_iter(supplier_id=supplier_framework["supplierId"]):
+            if user["active"]:
+                # generating ref separately so we can exclude certain parameters from the personalization dict
+                notify_ref = notify_client.get_reference(
+                    user["emailAddress"],
+                    notify_template_id,
+                    {
+                        "framework_slug": framework["slug"],
+                        "run_id": str(run_id),
+                    },
+                )
+                if dry_run:
+                    if notify_client.has_been_sent(notify_ref):
+                        logger.debug(
+                            "[DRY RUN] Would NOT send notification to {email_hash} (already sent)",
+                            extra={"email_hash": hash_string(user["emailAddress"])},
+                        )
+                    else:
+                        logger.info(
+                            "[DRY RUN] Would send notification to {email_hash}",
+                            extra={"email_hash": hash_string(user["emailAddress"])},
+                        )
+                else:
+                    try:
+                        notify_client.send_email(
+                            user["emailAddress"],
+                            notify_template_id,
+                            {
+                                "framework_name": framework["name"],
+                                "updates_url":
+                                    f"{get_web_url_from_stage(stage)}/suppliers/frameworks/{framework['slug']}/updates",
+                                "clarification_questions_open": framework["clarificationQuestionsOpen"],
```

Comment:
Is the plan here to use this context var to display optional content in the template? If so we might need to map to `yes`/`no` strings, as we will probably want to show `The deadline to ask clarification new questions has passed` if `clarification_questions_open` is False, and I think the Notify `((var??content))` syntax needs 'yes' or 'no' (I might be wrong).

The existing template

Reply:
I really left it open how we pass parameters to the template - what's here is little more than a sketch.

Reply:
OK. I'm happy to get a new template sorted out so we can start testing this and figure out what the Jenkins job will look like.

Reply:
I've added the new template now. I've called the optional var `clarification_questions_closed` so that we can do something like this:

```
clarification_questions_closed = "yes" if framework["clarificationQuestionsOpen"] else "no"
```

Reply:
Huzzah. Cool. Will update.